src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -444,33 +458,195 @@ static void _repeat(LottieGroup* parent, Shape* path, RenderContext* ctx)
+                    offsetPts.push(line.pt2);
+                    iPt++;
+                    break;
+                }
+
+                auto nextLine = cmds[iCmd + 1] == PathCommand::Close ? startLine : _offset(pts[iPt], pts[iPt + 1], offset);
+
+                bool inside{};
+                if (_intersect(line, nextLine, intersect, inside)) {
+                    if (inside) {
+                        if (cmds[iCmd + 1] == PathCommand::Close) offsetPts[startIndex] = intersect;
+                        offsetPts.push(intersect);
+                    }
+                    else {
+                        if (offsetPath->join == StrokeJoin::Round) {
+                            offsetPts.push(line.pt2);
+                            offsetCmds.push(PathCommand::CubicTo);
+                            offsetPts.push((line.pt2 + intersect) * 0.5f);
+                            offsetPts.push((nextLine.pt1 + intersect) * 0.5f);
+                            offsetPts.push(nextLine.pt1);
+                        } else {
+                            offsetPts.push(line.pt2);
+                            offsetCmds.push(PathCommand::LineTo);
+                            offsetPts.push(nextLine.pt1);
+                        }
+                    }
+                } //TODO: handle paraller
+                line = nextLine;
+
+                iPt++;
+                break;
+            }
+            case PathCommand::CubicTo: {
+                auto line1 = _offset(pts[iPt - 1], pts[iPt], offset);
+                auto line2 = _offset(pts[iPt], pts[iPt + 1], offset);
+                auto line3 = _offset(pts[iPt + 1], pts[iPt + 2], offset);
+
+                if (addStart) {
+                    offsetCmds.push(PathCommand::MoveTo);
+                    startIndex = offsetPts.count;
+                    offsetPts.push(line1.pt1);
+                    startLine = line;
+                    addStart = false;
+                }
+
+                offsetCmds.push(PathCommand::CubicTo);
+
+                bool inside{};
+                Point intersect{};
+                _intersect(line1, line2, intersect, inside);
+                offsetPts.push(intersect);
+                _intersect(line2, line3, intersect, inside);
+                offsetPts.push(intersect);
+                offsetPts.push(line3.pt2);
+
+                iPt += 3;
+                break;
+            }
+            case PathCommand::Close: {
+                offsetCmds.push(PathCommand::Close);
+                break;
+            }
+        }
+    }
+    shape->appendPath(offsetCmds.data, offsetCmds.count, offsetPts.data, offsetPts.count);
+}
+
+
+static void _appendRect(Shape* shape, float x, float y, float w, float h, float r, const RenderOffsetPath* offsetPath, Matrix* transform, bool clockwise)
 {
     //sharp rect
     if (tvg::zero(r)) {
-        PathCommand commands[] = {
+        //TODO: temporary changes - for consistency with other closed shapes
```

Comment:
I modified this to test _appendOffsetPath. rect is the only shape which is not closed with a lineTo before close is called.